lec01-introduction.Rmd

```diff
@@ -2,3 +2,171 @@
+
+```{r}
+summary(iris)
+```
+
+You can already see how it will save you time to use R instead of typing these calculations separately in Excel.
+ 
+## RStudio
+Before we continue, we are going to switch from using the R console, to using an interface called RStudio. RStudio includes the R console, but also many other convenient functionalities, which makes it easier to get started and to work with R. When you launch RStudio, you will see four panels:
+- The console is the same as what we have been using so far. The advantage with using the console within RStudio is that it interact with the other panels.
+- So, if I read in the iris data we were working with previously, we can view it in the console by typing `iris`, just like before, or we can also view the `iris` variable in our environment tab, which shows it to us in a spreadsheet or table format.
+- This panel shows the files in the current directory, any plots we might make later, and also the documentation, which we accessed earlier by typing `?` in front of a function. Here, the documentation is formatted in a way that is easier to read and also provides links to the related sections.
+- Finally we have the text editor panel. This is where we can write scripts, i.e. putting several commands of code together and saving them as a text document so that they are accessible for later and so that we can execute them all at once by running the script instead of typing them in one by one.
+
+So if I create a script with some of the commands you saw earlier, I can save them all as a text file and then run the entire script by clicking the run button, which executes commands in the console. We will use this text editor to explore how to make plots in R.
+
+## Plots
+In R, it is straightforward to make explore your data by creating nice looking plots. E.g. to make a scatter plot, we first tell R that we want to make a plot of the `iris` data frame, and we want the aesthetics of this plot to be that `sepal_length` is on the x axis and `sepal_width` is on the y-axis. Finally, we specify that we want this to be a plot of points, i.e. a scatter plot.
+
+```{r}
+ggplot(iris, aes(x=sepal_length, y=sepal_width)) +
+    geom_point()
+```
+
+This plots shows up in the plot panel in RStudio, which is convenient instead of having it show up as a separate window. You can already see that there seems to be at least some structure to the data. There is one cluster of points up the left corner and at least one in the lower right corner. Hmmmm, I wonder if these correspond to the different species of flowers... Let's find out, by coloring the points according to which species they are.
+
+```{r}
+ggplot(iris, aes(x=sepal_length, y=sepal_width, color=species)) +
+    geom_point()
+```
+
+Likewise, it would be really easy to make a histogram of the petal_length and color it according to the species.
```

Comment:
colour

Reply:
I am a traditionalist, so I stick with American English!

Reply:
That sounds like an oxymoron. We use Canadian English here... And anyway, you've got `ggpairs(iris, aes(colour = species))` already.

Reply:
It does! But it is actually not, the current American spelling is older. Before the Brits got all fancy about it... tsk Europeans.

Reply:
Well, that was an interesting detour, but I'm not sure that's true, unless you were speaking Latin.

Reply:
Interesting debate. It's true [color is older](http://www.etymonline.com/index.php?term=color&allowed_in_frame=0) and its not actually 'American'. It was popularized by an American though, to simplify spelling. Colour on the other hand comes from French, which makes sense why UK people use it since French and English mingled quite a bit.

Reply:
That page doesn't appear to give a claim as to the age of colour, though. Others show [multiple](https://www.grammarly.com/blog/color-colour/) [spellings](http://grammarist.com/spelling/color-colour/) in its early history. Alas, [ngrams](https://books.google.com/ngrams/graph?content=color%2Ccolour&year_start=1000&year_end=2000&corpus=15&smoothing=3&share=&direct_url=t1%3B%2Ccolor%3B%2Cc0%3B.t1%3B%2Ccolour%3B%2Cc0) is not old enough for this one.

Also, stop giving me [things](https://www.xkcd.com/386/) to procrastinate on. Have we argued about [this](https://www.xkcd.com/1285/) yet?

Reply:
Yes, exactly. Although, I didn't know that the original `color` spelling also came from French as the etymology dictionary indicates. Now the interesting question is, why did the French change to `colour`? Clearly the Brits just follow to appear classy =)